xtask/src/launcher.rs

```diff
@@ -105,28 +105,47 @@ fn build_released_binary(name: &str, directory: &str) -> Step {
 }
 
 pub fn run_launcher_test() -> Step {
+    let mut steps = vec![build_binary(
+        "build Oak Functions loader",
+        "./oak_functions_launcher",
+    )];
+    steps.extend(LauncherMode::iter().map(run_variant));
+
     Step::Multiple {
         name: "End-to-end tests for the launcher and enclave binary".to_string(),
-        steps: LauncherMode::iter().map(run_variant).collect(),
+        steps,
     }
 }
 
 fn run_variant(variant: LauncherMode) -> Step {
-    Step::Multiple {
-        name: format!("run {} variant", variant),
-        steps: vec![
+    let mut steps = vec![build_binary(
+        "build Oak Functions enclave binary",
+        &variant.enclave_crate_path(),
+    )];
+    steps.extend(match variant {
+        LauncherMode::Virtual => vec![
             build_stage0(),
-            build_binary("build Oak Functions loader", "./oak_functions_launcher"),
-            build_binary(
-                "build Oak Functions enclave binary",
-                &variant.enclave_crate_path(),
-            ),
-            Step::WithBackground {
-                name: "background loader".to_string(),
-                background: run_launcher(variant),
-                foreground: Box::new(run_client("test_key", "^test_value$", 300)),
+            build_binary("shim", "oak_enclave_shim"),
+            Step::Single {
+                name: "merge shim and enclave binary".to_string(),
```

Comment:
Can we document the main idea behind this `objcopy` call?

I think this is where the magic happens, or is it not? 

Can we call this outside `xtask` too, or can we have a `xtask` command which would do that e.g., `xtask build_enclave_binary`?

Reply:
I've added a comment as to what we exactly need to build here.

We _could_ have a separate xtask command, but my hope is that you'd never have to assemble the thing yourself by hand. For example, internally I'm planning to write a Starlark rule that does that objcopy step for you, so from the user's perspective is still "do a cargo build and use the resulting binary".
Thus, I'd like to put off adding a separate xtask command, at least for now.

Reply:
What I mean is that I hope you don't really have to care about the shim-binary split. But I suspect that in the longer run you're correct and we're going to need something like that.